test/kafo/hook_context_test.rb

```diff
@@ -85,5 +86,29 @@ module Kafo
         assert_equal false, context.app_option?('unknown_option')
       end
     end
+
+    describe "#has_custom_fact? with fact value present" do
+      let(:config) { Minitest::Mock.new }
+      let(:custom_fact_storage) { { 'my_custom_fact' => 'my_custom_fact_value' } }
+
+      specify do
+        kafo.expect :config, config
+        config.expect :has_custom_fact?, custom_fact_storage
```

Comment:
I think based on the failure this needs to be:

```
config.expect :has_custom_fact?, custom_fact_storage, ['my_custom_fact']
```

Reply:
So if I understand this correctly, need to declare that the mocked function expects the fact name as an argument. I'll give this a try. Thanks @ehelms